### ISSUE TITLE
Some unit tests do not pass locally, but pass in bazel CI

### ISSUE THREAD

**Issue by liggitt** [kubernetes/kubernetes#78982]
**What happened**:

Some unit tests are not passing:

```
go test ./vendor/k8s.io/component-base/cli/flag
--- FAIL: TestConstantMaps (0.01s)
    ciphersuites_flag_test.go:106: discovered version tls.VersionTLS13 not in version map
FAIL
FAIL	k8s.io/kubernetes/vendor/k8s.io/component-base/cli/flag	0.312s

go test ./vendor/k8s.io/apiserver/pkg/storage/value/encrypt/envelope
--- FAIL: TestIntermittentConnectionLoss (3.01s)
    grpc_service_unix_test.go:194: Connected to KMSPlugin
    grpc_service_unix_test.go:197: KMS Plugin is stopping
    grpc_service_unix_test.go:206: Sending encrypt request
    grpc_service_unix_test.go:219: failed to start test KMS provider server, error: failed to listen on the unix socket, error: listen unix @753031d5-2aa9-4b77-95d6-2e4614c4d348.sock: bind: address already in use
panic: Fail in goroutine after TestIntermittentConnectionLoss has completed
```

Yet CI unit test jobs (both merge-blocking and post-merge) are passing.

**What you expected to happen**:

CI to run all unit tests

/cc @kubernetes/sig-testing @sttts 
/priority critical-urgent
/milestone v1.15

**Comment by liggitt** [comment]
https://github.com/kubernetes/kubernetes/pull/78978 adds the missing packages to some of the test invocations, but pull-kubernetes-bazel-test is still green on that PR, which means bazel test is using some other method for finding unit tests to run

**Comment by liggitt** [comment]
good news on the ciphersuites test, it looks like it is being run, but the test assumed it could parse go source, which is not accurate in a bazel env, and passed if it failed looking up the source

**Comment by liggitt** [comment]
also good news on the grpc test, it looks like that passes reliably on linux, which is where bazel is running

**Comment by liggitt** [comment]
summary: the two failing tests are explainable as:
* bazel ci doesn't have go source available and the ciphersuites test decided to skip and report success in that case (fix for the bug in https://github.com/kubernetes/kubernetes/pull/78984 needs to be in 1.15, rework of the test will happen in 1.16)
* bazel ci runs the envelope grpc tests on linux where they pass reliably, but the tests make assumptions about abstract sockets not leaving around files which do not hold true for other unix platforms like darwin (fix for the test can be in 1.16)

lowering priority and moving out of the milestone

/milestone v1.16
/priority important-soon

**Comment by BenTheElder** [comment]
> also good news on the grpc test, it looks like that passes reliably on linux, which is where bazel is running

and _all_ of our CI, except some limited windows node e2e stuff ...

**Comment by BenTheElder** [comment]
> bazel ci doesn't have go source available and the ciphersuites test decided to skip and report success in that case (fix for the bug in #78984 needs to be in 1.15, rework of the test will happen in 1.16 in #78986)

er meant to also comment that we can make source available to tests with data dependencies specifying a file group with the sources. looks like I only mentioned it here https://github.com/kubernetes/kubernetes/pull/78986#issuecomment-503080982

**Comment by xmudrii** [comment]
@liggitt @BenTheElder Hello! I'm bug triage lead for the 1.16 release cycle and considering this issue is tagged for 1.16, but not updated for a long time, I'd like to check its status. The code freeze is starting on August 29th (about 1.5 weeks from now), which means that there should be a PR ready (and merged) until then.

Do we still target this issue to be fixed for 1.16?

**Comment by BenTheElder** [comment]
I'll leave this one to @liggitt
